RocketChat/Rocket.Chat
# feat: Add setting to synchronize LDAP info on OAuth logins

## Proposed changes (including videos or screenshots)


 - Add a "Update User Data on Login with OAuth services" setting under LDAP Data Sync settings to enable users to have their data synced with LDAP servers on login with OAuth services.



## Issue(s)


## Steps to test or reproduce

Go to **Workspace > Settings > LDAP > Data Sync** and enable **"Update User Data on Login"** and **"Update User Data on Login with OAuth services"**. Then try to login using an OAuth service. The logged in user should be synced with its LDAP server corresponding registry (the username should match) on login

## Further comments

TC-433
